extensions/blocks/videopress/edit.js

```diff
@@ -90,52 +109,158 @@ const VideoPressEdit = CoreVideoEdit =>
 			} );
 		};
 
+		onSelectPoster = image => {
+			const { setAttributes } = this.props;
+			setAttributes( { poster: image.url } );
+		};
+
 		onRemovePoster = () => {
-			this.props.setAttributes( { poster: '' } );
+			const { setAttributes } = this.props;
+			setAttributes( { poster: '' } );
 
 			// Move focus back to the Media Upload button.
 			this.posterImageButton.current.focus();
 		};
 
+		toggleAttribute = attribute => {
+			return newValue => {
+				this.props.setAttributes( { [ attribute ]: newValue } );
+			};
+		};
+
+		getAutoplayHelp = checked => {
+			return checked
+				? __( 'Note: Autoplaying videos may cause usability issues for some visitors.', 'jetpack' )
+				: null;
+		};
+
 		render() {
 			const {
 				attributes,
 				className,
+				instanceId,
 				isFetchingPreview,
 				isSelected,
 				isUploading,
 				preview,
 				setAttributes,
 			} = this.props;
 			const { fallback, isFetchingMedia } = this.state;
+			const { autoplay, caption, controls, loop, muted, poster, preload } = attributes;
+
+			const videoPosterDescription = `video-block__poster-image-description-${ instanceId }`;
+
+			const blockSettings = (
+				<Fragment>
+					<BlockControls>
+						<Toolbar>
+							<IconButton
+								className="components-icon-button components-toolbar__control"
+								label={ __( 'Edit video', 'jetpack' ) }
+								onClick={ this.switchToEditing }
+								icon="edit"
+							/>
+						</Toolbar>
+					</BlockControls>
+					<InspectorControls>
+						<PanelBody title={ __( 'Video Settings', 'jetpack' ) }>
+							<ToggleControl
+								label={ __( 'Autoplay', 'jetpack' ) }
+								onChange={ this.toggleAttribute( 'autoplay' ) }
+								checked={ autoplay }
+								help={ this.getAutoplayHelp }
+							/>
+							<ToggleControl
+								label={ __( 'Loop', 'jetpack' ) }
+								onChange={ this.toggleAttribute( 'loop' ) }
+								checked={ loop }
+							/>
+							<ToggleControl
+								label={ __( 'Muted', 'jetpack' ) }
+								onChange={ this.toggleAttribute( 'muted' ) }
+								checked={ muted }
+							/>
+							<ToggleControl
+								label={ __( 'Playback Controls', 'jetpack' ) }
+								onChange={ this.toggleAttribute( 'controls' ) }
+								checked={ controls }
+							/>
+							<SelectControl
+								label={ __( 'Preload', 'jetpack' ) }
+								value={ preload }
+								onChange={ value => setAttributes( { preload: value } ) }
+								options={ [
+									{ value: 'auto', label: __( 'Auto', 'jetpack' ) },
+									{ value: 'metadata', label: __( 'Metadata', 'jetpack' ) },
+									{ value: 'none', label: __( 'None', 'jetpack' ) },
+								] }
+							/>
+							<MediaUploadCheck>
+								<BaseControl
+									className="editor-video-poster-control"
+									label={ __( 'Poster Image', 'jetpack' ) }
```

Comment:
Should this be lowercase "Poster image" instead?

Reply:
Since the extension fallbacks to the core video block I preferred to use the same wording used there, so the text doesn't change depending on VideoPress being enabled.

https://github.com/WordPress/gutenberg/blob/467e45f787f93d35030ce266a500c627db0e18e5/packages/block-library/src/video/edit.js#L260

Reply:
Makes sense! 👍